chain-abstraction/sol-implementation/edge_challenge_manager.go

```diff
@@ -391,19 +387,21 @@ func (e *specEdge) TopLevelClaimHeight(ctx context.Context) (protocol.OriginHeig
 
 // Wrapper around the challenge manager contract with developer-friendly methods.
 type specChallengeManager struct {
-	addr           common.Address
-	backend        protocol.ChainBackend
-	assertionChain *AssertionChain
-	txOpts         *bind.TransactOpts
-	caller         *challengeV2gen.EdgeChallengeManagerCaller
-	writer         *challengeV2gen.EdgeChallengeManagerTransactor
-	filterer       *challengeV2gen.EdgeChallengeManagerFilterer
+	addr                  common.Address
+	backend               protocol.ChainBackend
+	assertionChain        *AssertionChain
+	txOpts                *bind.TransactOpts
+	caller                *challengeV2gen.EdgeChallengeManagerCaller
+	writer                *challengeV2gen.EdgeChallengeManagerTransactor
+	filterer              *challengeV2gen.EdgeChallengeManagerFilterer
+	challengePeriodBlocks uint64
+	numBigStepLevel       uint8
 }
 
 // NewSpecChallengeManager returns an instance of the spec challenge manager
 // used by the assertion chain.
 func NewSpecChallengeManager(
```

Comment:
The way that we should probably initialize a challenge manager is by looking at the `challengeManager()` method of the rollup contracts via Solidity bindings. Maybe we need to do so by assertion hash? That is, if you pass in assertion hash A which has challenge manager address X, and assertion hash B, which has challenge manager address Y, these bindings should be initialized for the respective address

Reply:
Resolving based on offline discussion.